### PR TITLE
AddTasking RSP syntax to printCompdb functionality

### PR DESCRIPTION
**Pull request by gerioldman**
This PR solves #2313

### CODE REVIEW

File: src/ninja.cc [hunk 1]
```diff
@@ -892,7 +892,12 @@ std::string EvaluateCommandWithRspfile(const Edge* edge,
     rspfile_content.replace(newline_index, 1, 1, ' ');
     ++newline_index;
   }
-  command.replace(index - 1, rspfile.length() + 1, rspfile_content);
+  if (command[index - 1] == '@')
```
Comment (jhasse) [inline src/ninja.cc:895]: Please add {}, see https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching

**Review by jhasse — CHANGES_REQUESTED** [review]

File: src/ninja.cc (hunk 1 above)
Comment (gerioldman) [reply src/ninja.cc]: I used clang-format for this as well, hope that's fine now

File: src/ninja.cc [hunk 2]
```diff
@@ -892,7 +892,12 @@ std::string EvaluateCommandWithRspfile(const Edge* edge,
...
+    command.replace(index - 1, rspfile.length() + 1, rspfile_content);
+  else if (command.find("-f ") == index - 3)
+    command.replace(index - 3, rspfile.length() + 3, rspfile_content);
+  else
```
[the remark this answers is not in the document]
Comment (gerioldman) [reply src/ninja.cc]: Added a comment about that

File: src/ninja.cc [hunk 3]
```diff
@@ -882,7 +882,7 @@ std::string EvaluateCommandWithRspfile(const Edge* edge,
     return command;
 
   size_t index = command.find(rspfile);
-  if (index == 0 || index == string::npos || command[index - 1] != '@')
+  if (index == 0 || index == string::npos || ( command[index - 1] != '@' && command.find("--option-file=") != index - 14 && command.find("-f ") != index - 3 ))
```
[the remark this answers is not in the document]
Comment (gerioldman) [reply src/ninja.cc]: Ran clang-format, is the multiline expression the expected result?

[the remark this answers is not in the document]
Comment (jhasse) [reply src/ninja.cc]: yes :)

**Pull request merged**